### DISCUSSION TITLE
如何使用 hugo-theme-virgo 主题

### DISCUSSION THREAD

**Discussion by giscus** [loveminimal/comment#17] [General]
# 如何使用 hugo-theme-virgo 主题

如何使用 hugo-theme-virgo 主题 - https://walkssi.com/%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8-hugo-theme-virgo-%E4%B8%BB%E9%A2%98/

https://walkssi.com/%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8-hugo-theme-virgo-%E4%B8%BB%E9%A2%98/

**Comment by friday-ckp** [comment]
放了nav.md 还是没有正常显示首页, 不知道啥原因, 我复制粘贴的

**Reply by loveminimal** [reply]
![image](https://github.com/loveminimal/comment/assets/23427211/0761b2c4-8741-4bbd-8ddd-4de51fa29870)

按上图格式试一下。

**Comment by happyzhangyyds** [comment]
作者大大有打算适配twikoo插件嘛，感觉twikoo也蛮好用的。

**Reply by loveminimal** [reply]
目前没这个计划哈，定位就是个纯前端的静态博客，平时写写画画。😄

**Comment by x1any** [comment]
麻烦请教一下，导航按钮旁边的“Ship”按钮如何去掉呢

**Reply by loveminimal** [reply]
当前版本在配置项中是无法去掉的。
如果你真的想去掉，可以尝试去修改源码，如下：
https://github.com/loveminimal/hugo-theme-virgo/blob/master/layouts/partials/_contact.html
删除 `_contact.html` 文件的 10~12 行代码，即可。

**Reply by loveminimal** [reply]
最新的版本已经添加了配置项，如下：
`config.toml` 中的 76 行 `ship = '你的链接' `

**Reply by x1any** [reply]
十分感谢！

**Reply by x1any** [reply]
另外我发现一个bug，小红书icon点击跳转到https://www.xiaohongshu.com/explore而不是我设定的个人页面，是不是小红书的限制啊，我看博主的小红书也是

**Reply by loveminimal** [reply]
我知道是什么问题了，如下 ：
小红书的个人页，默认域名是 `www.xiaohongshu.com/user/profile/xxx` ，
当你使用 `xiaohongshu.com/user/profile/xxx` 访问时，、
它并没有在保持原域名路径的情况下重定向，
而是粗爆的重定向到 explore 页。

如果，你想解决这个问题，需要修改
https://github.com/loveminimal/hugo-theme-virgo/blob/master/layouts/partials/_contact.html
`_contact.html` 文件 78行，在 `xiaohongshu.com` 前添加 `www.` 即可。

**Reply by loveminimal** [reply]
最新版本已修复~

**Comment by Chaaoo** [comment]
老哥，旧版的search，about怎么写？我咋写都不对。

**Reply by loveminimal** [reply]
就和正常创建文章的 markdown 一样~

```
title: 'xxx'
```

title 是必须的，因为要依赖它生成对应的路由地址。

**Comment by naner34** [comment]
请问旧版本的搜索关键字，明明有文章有这个关键字，但搜索不到内容是怎么回事呢？

**Reply by loveminimal** [reply]
这个真没注意过，按理说不应该存在此类状况。
旧版的搜索方式是实时的，内容更新后应该可以搜到了。

**Reply by naner34** [reply]
更新了，应该不是这个问题，点进去搜索，输入关键字后没反应就算了，点击左边的X也没法清空已输入内容，感觉像代码没起作用。我重新新建site，下载旧主题试还是这样。
我描述一下自己下载主题后的操作：
因为themes\virgo\layouts\partial里面有_search.html，themes\virgo\static\js里面有.search.js，然后我就 'hugo new content search.md'，里面 的fontmatter 只保留了title和date【title: "Search"，date: 2024-03-16】，然后就 hugo server 看搜索功能有没有起效。
说实话第一次这样操作是成功的【在上午】，但下午再试试的时候就没用了，换了浏览器也不起效哦。

**Reply by loveminimal** [reply]
确实出现了此类问题，是由搜索索引内容不完全造成的，后续有时间看一下。

**Reply by loveminimal** [reply]
完整版的搜索问题已经修复，你可以在分支 1.0.0 获取到最近的更新~
https://github.com/loveminimal/hugo-theme-virgo/tree/1.0.0
或者，你只是关心作了哪些修改，更改如下：
https://github.com/loveminimal/hugo-theme-virgo/commit/c231e5cb9e9f16d5c92a64625e88ec777ff4d3ac

**Reply by naner34** [reply]
我又来了，哈哈，我发现能搜索了，但是好像显示的搜索结果里面，相同的文章会出现2遍以上。QAQ
注：我的操作是把旧的主题文件夹内的static和layout覆盖了之前的旧版。

**Reply by loveminimal** [reply]
已修复，见最近提交~

PS:（兄弟，歇一会儿，我有强迫症，看见 bug 就想修复.... 
你再发现问题了，过两天再提交，最近流感中标，难受ing....）

**Comment by tarkyn915** [comment]
博主你好，我非常喜欢你设计的hugo主题，但是我有几个问题想要请教一下
关于社交媒体的界面，能否自定义设置链接呢，而不是固定式跳转的方式，因为我看tg的跳转好像有点问题
还有就是能否自定义添加社交媒体，我想添加一个discord但是不行
我是一个小白，我对网页这块知识并不是很充足，如有打扰，请见谅，下面我附上我之前用的博客ui方案，我感觉还不错很方便简单

https://github.com/adityatelange/hugo-PaperMod/wiki/Icons

```
# Social icons
[[params.social]]
name = "telegram"
url = "https://t.me/xxxxx"

[[params.social]]
name = "discord"
url = "https://discord.gg/MgJf4DJvdn"
```

**Reply by loveminimal** [reply]
很高兴你喜欢它~

目前不支持自定义图标和链接，主要原因在于图标的大小（svg格式）导致的。  
直接修改了 svg 中的属性，以使得图标的显示比较合理美观。  

后续有时间，有比较合适的图标库了，  
可以调整一下，  添加一个允许如你描述的这种自定义方式。  

如果，你现在想自定义链接的话，  你不得不修改源码了。

**Reply by loveminimal** [reply]
> 博主你好，我非常喜欢你设计的hugo主题，但是我有几个问题想要请教一下 关于社交媒体的界面，能否自定义设置链接呢，而不是固定式跳转的方式，因为我看tg的跳转好像有点问题 还有就是能否自定义添加社交媒体，我想添加一个discord但是不行 我是一个小白，我对网页这块知识并不是很充足，如有打扰，请见谅，下面我附上我之前用的博客ui方案，我感觉还不错很方便简单
> 
> https://github.com/adityatelange/hugo-PaperMod/wiki/Icons
> 
> ```
> # Social icons
> [[params.social]]
> name = "telegram"
> url = "https://t.me/xxxxx"
> 
> [[params.social]]
> name = "discord"
> url = "https://discord.gg/MgJf4DJvdn"
> ```

最新的更新已经添加了 discord ，  
telegram 已经更改为 https://t.me/xxxx 形式

**Reply by tarkyn915** [reply]
哇 非常感谢博主的更新！
但是我测试了一下tg的链接确实正常了，discord的固定链接格式好像有点问题，无论是填写个人id或是群组的链接都没办法正常跳转，所以感觉还是自定义social的链接会比较好，希望后续能优化
另外我还发现一个问题，就是主页显示文章数量好像不准确，他检测的是content 里的数量？ 我在content里添加一个文件夹一个md文件 他就+1  而不是检测posts里的文章数量
ps 整体感觉很不错了，就是我也是一个强迫症，遇到一些小瑕疵就感觉很难受 /(ㄒoㄒ)/~~

**Reply by loveminimal** [reply]
> 哇 非常感谢博主的更新！ 但是我测试了一下tg的链接确实正常了，discord的固定链接格式好像有点问题，无论是填写个人id或是群组的链接都没办法正常跳转，所以感觉还是自定义social的链接会比较好，希望后续能优化 另外我还发现一个问题，就是主页显示文章数量好像不准确，他检测的是content 里的数量？ 我在content里添加一个文件夹一个md文件 他就+1 而不是检测posts里的文章数量 ps 整体感觉很不错了，就是我也是一个强迫症，遇到一些小瑕疵就感觉很难受 /(ㄒoㄒ)/~~

可以自定义地址了~

**Comment by atopos-x** [comment]
博主大哥，在hugo的hugo.toml配置文件中enableGitInfo = true这个属性这样子配置的的话编译会报错.git文件找不到，但是如果时false的话就正常编译，如果您看到了我的消息，麻烦回复一下谢谢大哥，我查了相关的一些东西，但是都没找到解决办法。哎！求帮助

**Reply by atopos-x** [reply]
还有github图片无法显示的问题，在还没有打开文章的时候，在文章外可以正常看到，但带点进去就看不到了，只能看到一个很小的缩略，是需要加速网站还是怎么弄?

**Reply by loveminimal** [reply]
> 博主大哥，在hugo的hugo.toml配置文件中enableGitInfo = true这个属性这样子配置的的话编译会报错.git文件找不到，但是如果时false的话就正常编译，如果您看到了我的消息，麻烦回复一下谢谢大哥，我查了相关的一些东西，但是都没找到解决办法。哎！求帮助

是不是博客内容没有使用 Git 管理起来呢？在与 `config.toml` 同级的目录中需要有 `.git` 这个文件夹，默认是隐藏的~

github 图片无法显示，需要看一下资源目录的路径是不是对的~

**Reply by atopos-x** [reply]
博主我想问一下，就是我使用的是图片都放在assets文件夹中这种结构的，但是编译之后，照片的路径还是和文章同级别的，并不会编译成assets下的照片路径，这是新版本的功能吗？然后老版本只能老老实实一个文件夹，然后里面是index.md和一个assets这种结构吗？

**Reply by loveminimal** [reply]
> 博主我想问一下，就是我使用的是图片都放在assets文件夹中这种结构的，但是编译之后，照片的路径还是和文章同级别的，并不会编译成assets下的照片路径，这是新版本的功能吗？然后老版本只能老老实实一个文件夹，然后里面是index.md和一个assets这种结构吗？

你在文章中引用图片的时候，路径要用类似路径才可以，如 `assets/nuli_cr1.jpg` ，而不能是 `./assets/nuli_cr1.jpg` 这种。

**Reply by atopos-x** [reply]
我主题用的是旧的，然后图片是按照你教程上的使用Obsidian放置图片的位置，路径就是assets/nuli_cr1.jpg，但是编译之后，我开发者模式看，src的值，前半部分基本上都是到当前文件的名然后拼接上照片路径，就会是这样，posts/aaa/assets/nuli_cr1.jpg，但是我的aaa是一个.md文件，并且和assets同级。

**Reply by loveminimal** [reply]
不好意思，我好像明白你说的这种情况了~

制作旧版本主题的时候，我日常是使用 VSCode 和 Typora 管理 `.md` 文件的，所以那个时候为了方便管理图片，比如删除文章，图片也可以一起删除等，就把文章和图片一起放在了同一目录下。

后来使用 Obsidian ，它有一个插件可以一键删除项目内没有被引用的图片，所有就把图片全放在 `assets` 中了。后期，用  Hugo 编译后，为了解决类似 `[[ xxx ]]` 这种 wiki 格式的链接和图片格式，使用 JS 做了统一的处理，比如在每个链接前面加了一个 `/` ，基本上就是这个逻辑导致的。

目前如果你继续使用旧版本的主题的话，建议你使用 markdown 本身语法支持的那种链接和图片格式，即 `[xxx](xxx)` 这种。使用新版本主题的话，使用类 wiki 的链接格式是没有什么问题的。

**Reply by atopos-x** [reply]
好的，感谢博主解答

**Comment by xgnw** [comment]
大佬，我是小白，安装hugo设置主题搞了几天还是跟你的站不一样，只能修改hugo.toml这个文件，但是把你的设置放进去就报错，能不能像这个主题一样，https://github.com/hugo-next/hugo-theme-next-starter设置一个一键安装的，对我们小白比较友好

**Reply by loveminimal** [reply]
原来在邮件中回复，这边是看不到的……
这个只有等有时间再弄了，这段时间好忙~

**Comment by atopos-x** [comment]
博主你好，我想问一下旧版本首页的导航栏正常都是一个xxx.com/nav这种结构，我想添加一个导航栏的按钮，点击后跳转到指定的链接，而不是一个nva的一个页面，怎么配置？

**Reply by loveminimal** [reply]
https://github.com/loveminimal/hugo-theme-virgo/blob/48a2d7d04f526c510bd18253387fd1a6d9e79353/config/config-old.toml#L62  
在站点的 `config.toml` 中，将对应的 `path` 直接设置为你想要的链接就可以了，如 `path = 'https://baidu.com'`。

**Reply by atopos-x** [reply]
好的，麻烦问一下博主，旧主题主页的猫的svg图片挺好看，看起来感觉质感很好，是自己制作的，还是在第三方网站找的，如果是网站，可以分享一下吗？感谢🙏

**Reply by loveminimal** [reply]
https://github.com/loveminimal/hugo-theme-virgo/blob/1.x/static/imgs/cat.svg

使用的话，这里可以直接下载的 ~  
我也忘记当时是从哪里找的了~

**Reply by atopos-x** [reply]
好的，感谢🙏

**Comment by richbridge** [comment]
博主你好，求助：我搭建的新版主题，导航页没有本地搜索应该如何解决？谢谢。

**Reply by loveminimal** [reply]
> 博主你好，求助：我搭建的新版主题，导航页没有本地搜索应该如何解决？谢谢。
![image](https://github.com/user-attachments/assets/8167f6a7-6d84-4bae-ae53-9e9f5a057b63)

站点配置文件中，这一行需要取消注释，修改为 `true`

🪧 原来是我忘记更新主题中的配置文件了，现在已经推送了，你看一下最新的提交就可以了。
